code/datums/elements/backblast.dm

```diff
@@ -0,0 +1,74 @@
+/**
+  * When attached to a gun and the gun is successfully fired, this element creates a "backblast" of fire and pain, like you'd find in a rocket launcher or recoilless rifle
+  *
+  * The backblast is simulated by a number of fire plumes, or invisible incendiary rounds that will torch anything they come across for a short distance, as well as knocking
+  * back nearby items.
+  */
+/datum/element/backblast
+	element_flags = ELEMENT_BESPOKE
+	id_arg_index = 2
+
+	/// How many "pellets" of backblast we're shooting backwards, spread between the angle defined in angle_spread
+	var/plumes
+	/// Assuming we don't just have 1 plume, this is the total angle we'll cover with the plumes, split down the middle directly behind the angle we fired at
+	var/angle_spread
+	/// How far each plume of fire will fly, assuming it doesn't hit a mob
+	var/range
+
+/datum/element/backblast/Attach(datum/target, plumes = 4, angle_spread = 48, range = 8)
+	. = ..()
+	if(!isgun(target) || plumes < 1 || angle_spread < 1 || range < 1)
+		return ELEMENT_INCOMPATIBLE
+
+	src.plumes = plumes
+	src.angle_spread = angle_spread
+	src.range = range
+
+	if(plumes == 1)
+		RegisterSignal(target, COMSIG_GUN_FIRED, .proc/gun_fired_simple)
+	else
+		RegisterSignal(target, COMSIG_GUN_FIRED, .proc/gun_fired)
+
+/datum/element/backblast/Detach(datum/source, force)
+	if(source)
+		UnregisterSignal(source, COMSIG_GUN_FIRED)
+	. = ..()
+
+/// For firing multiple plumes behind us, we evenly spread out our projectiles based on the [angle_spread][/datum/element/backblast/var/angle_spread] and [number of plumes][/datum/element/backblast/var/plumes]
+/datum/element/backblast/proc/gun_fired(obj/item/gun/weapon, mob/living/user, atom/target, params, zone_override)
+	SIGNAL_HANDLER
+
+	if(!weapon.chambered || HAS_TRAIT(user, TRAIT_PACIFISM))
```

Comment:
what about golems and such, or not meeting the firing pin requirements of the gun?